qemu/qmp.go

```diff
@@ -1255,6 +1255,46 @@ func (q *QMP) ExecHotplugMemory(ctx context.Context, qomtype, id, mempath string
 	return err
 }
 
+// ExecuteNVDIMMDeviceAdd adds the guest portion of a block device to
+// a QEMU instance using a NVDIMM driver with the device_add command.
+// id is the id of the device to add.  It must be valid QMP identifiers.
+// mempath is the path of the device to add, e.g., /dev/rdb0.  size is
+// the data size of the device.
+func (q *QMP) ExecuteNVDIMMDeviceAdd(ctx context.Context, id, mempath string, size int64) error {
+	args := map[string]interface{}{
+		"qom-type": "memory-backend-file",
+		"id":       "nvdimmbackmem" + id,
+		"props": map[string]interface{}{
+			"mem-path": mempath,
+			"size":     size,
+			"share":    true,
+		},
+	}
+	err := q.executeCommand(ctx, "object-add", args, nil)
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		if err != nil {
+			q.cfg.Logger.Errorf("Unable to hotplug NVDIMM device: %v", err)
+			err = q.executeCommand(ctx, "object-del", map[string]interface{}{"id": "nvdimmbackmem" + id}, nil)
```

Comment:
Doesn't this overwrite the error code from q.executeCommand(ctx, "device_add", args, nil)?  Assuming this command succeeds we'll return nil even though there was an error.  Probably best just to log any error from object-del and ignore it.